etc/otel-jmx.config.yaml

```diff
@@ -43,31 +63,89 @@ rules:
        type: gauge
        desc: total records consumed
        unit: '{records}'
- - bean: kafka.streams:type=stream-task-metrics,thread-id=*,task-id=*
+
+ # Kafka Streams thread metrics
+ - bean: kafka.streams:type=stream-thread-metrics,thread-id=*
    metricAttribute:
      thread: param(thread-id)
-     task: param(task-id)
    mapping:
      process-total:
-       metric: kafka.streams.task.process.total
+       metric: kafka.streams.thread.process.total
        type: gauge
        desc: total records processed
        unit: '{records}'
- - bean: kafka.streams:type=stream-thread-metrics,thread-id=*
-   metricAttribute:
-     thread: param(thread-id)
-   mapping:
+     process-rate:
+       metric: kafka.streams.thread.process.rate
+       type: gauge
+       desc: rate of records processed
+       unit: '{records-per-second}'
+     process-ratio:
+       metric: kafka.streams.thread.process.ratio
+       type: gauge
+       desc: ratio of time spent processing
+     commit-rate:
+       metric: kafka.streams.thread.commit.rate
+       type: gauge
+       desc: rate of commit
+       unit: '{ops-per-second}'
+     commit-ratio:
+       metric: kafka.streams.thread.commit.ratio
+       type: gauge
+       desc: ratio of time spent committing
+     commit-latency-avg:
+       metric: kafka.streams.thread.commit.latency.avg
+       type: gauge
+       desc: avg latency of a commit
+       unit: '{milliseconds}'
+     commit-latency-max:
+       metric: kafka.streams.thread.commit.latency.max
+       type: gauge
+       desc: max latency of a commit
+       unit: '{milliseconds}'
+     poll-rate:
+       metric: kafka.streams.thread.poll.rate
+       type: gauge
+       desc: rate of poll calls
+       unit: '{ops-per-second}'
+     poll-ratio:
+       metric: kafka.streams.thread.poll.ratio
+       type: gauge
+       desc: ratio of time spent polling
+     poll-records-avg:
+       metric: kafka.streams.thread.poll.records.avg
+       type: gauge
+       desc: avg records per poll
+       unit: '{records}'
+     punctuate-ratio:
+       metric: kafka.streams.thread.punctuate.ratio
+       type: gauge
+       desc: ratio of time spent punctuating
+     task-closed-rate:
+       metric: kafka.streams.thread.task.closed.rate
+       type: gauge
+       desc: rate of tasks closed
+       unit: '{tasks-per-second}'
      blocked-time-ns-total:
        metric: kafka.streams.thread.blocked.time.total.ns
        type: gauge
        desc: total time the stream thread was blocked
        unit: '{nanoseconds}'
- - bean: kafka.streams:type=stream-thread-metrics,thread-id=*
+     thread-start-time:
+       metric: kafka.streams.thread.start.time
+       type: value
```

Comment:
Is this a valid type? Not sure what the options are, but this isn't a "gauge" unless that has a different meaning here -- it's not like this will be changing

Reply:
no - I think we can just use the gauge type.